src/arkode/arkode_arkstep.c

```diff
@@ -294,23 +286,22 @@ int ARKStepResize(void *arkode_mem, N_Vector y0, realtype hscale,
 
     /* destroy existing NLS object */
     retval = SUNNonlinSolFree(step_mem->NLS);
+    SUNCheckCallNoRet(retval);
     if (retval != ARK_SUCCESS)  return(retval);
```

Comment:
Does this check the same error twice, and could it be simplified to
```suggestion
    SUNCheckCall(SUNNonlinSolFree(step_mem->NLS));
```

Reply:
This would necessitate changing the return type of the function to a `SUNErrCode` or accepting `SUNErrCode` and `int` as interchangeable (from the compiler point-of-view they are, but semantically they are not really).